src/revwalk.c

```diff
@@ -756,15 +756,11 @@ int git_revwalk_add_hide_cb(
 	if (walk->walking)
 		git_revwalk_reset(walk);
 
-	if (walk->hide_cb) {
-		/* There is already a callback added */
-		giterr_set(GITERR_INVALID, "there is already a callback added to hide commits in revwalk");
-		return -1;
-	}
-
 	walk->hide_cb = hide_cb;
 	walk->hide_cb_payload = payload;
-	walk->limited = 1;
+
+	if (hide_cb)
+		walk->limited = 1;
```

Comment:
Mh. We don't zero out the struct members, so you'd probably also have to unlimit the revwalk in case there is no `hide_cb`. So:

```
walk_limited = !!hide_cb;
```